### PR TITLE
Refine v16 migration guide

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

None, as it's a documentation fix.

> Is there anything in the PR that needs further explanation?

This PR aligns the v16 migration with our style from the previous ones by:

- listing the changes by impactfulness
- leading with details of the change
- then explaining how to migrate

### CODE REVIEW

File: docs/migration-guide/to-16.md [hunk 1]
```diff
@@ -1,33 +1,58 @@
...
+We've changed the CLI to print problems to stderr instead of stdout.
+
+If you use the [`--fix`](../user-guide/cli.md#--fix) and [`--stdin`](../user-guide/cli.md#--stdin) options, the CLI will print the fixed code to stdout and any problems to stderr.
+
+## Changed CLI exit code for flag errors
+
+We've changed the exit code for CLI flag errors from `2` to `64` so that `2` is only used for lint problems.
 
-If you use `stylelint.lint()` to lint files, then the `code` property will always be `undefined`.
+If you're an author of an editor integration that uses the CLI, you can now distinguish between flag errors and lint problems.
```
Comment (Mouvedia) [inline docs/migration-guide/to-16.md:58]: Maybe we need to sync the releases of certain extensions with v16?
e.g. coordinate with @ota-meshi for stylelint/vscode-stylelint

Comment (ybiquitous) [reply docs/migration-guide/to-16.md:58]: In fact, it seems difficult since we have not maintained `vscode-stylelint` well recently. 😓

**Review by ybiquitous — APPROVED** [review]
Much clearer! LGTM 👍🏼

File: docs/migration-guide/to-16.md (hunk 1 above)
Comment (ota-meshi) [reply docs/migration-guide/to-16.md:58]: Yeah, I think we need to change vscode-stylelint.

**Pull request merged**